src/main/java/jnr/ffi/util/BufferUtil.java

```diff
@@ -64,7 +66,8 @@ public static CharSequence getCharSequence(ByteBuffer buf, Charset charset) {
         final ByteBuffer buffer = buf.slice();
         // Find the NUL terminator and limit to that, so the
         // StringBuffer/StringBuilder does not have superfluous NUL chars
-        int end = indexOf(buffer, (byte) 0);
+        final byte[] nullCharBytes = new byte[StringUtil.terminatorWidth(charset)];
```

Comment:
Perhaps we can cache the three known lengths and not reallocate this byte[] every time?